src/MLOps.NET/Storage/Repositories/DataRepository.cs

```diff
@@ -71,5 +73,67 @@ public Data GetData(Guid runId)
 
             return data;
         }
+
+        ///<inheritdoc cref="IDataRepository"/>
+        public async Task LogDataDistribution<T>(Guid runId, IDataView dataView, string columnName) where T : struct
+        {
+            using var db = this.contextFactory.CreateDbContext();
+
+            var data = db.Data
+                .Include(d => d.DataSchema.DataColumns)
+                .First(d => d.RunId == runId);
+
+            var dataColumn = data.DataSchema
+                .DataColumns
+                .FirstOrDefault(c => c.Name == columnName);
+
+            var column = dataView.Schema.First(c => c.Name == columnName);
+
+            List<DataDistribution> list = GetDataDistributionForColumn<T>(dataView, column, dataColumn);
+
+            db.DataDistributions.AddRange(list);
+
+            await db.SaveChangesAsync();
+        }
+
+        ///<inheritdoc cref="IDataRepository"/>
+        public List<DataDistribution> GetDataDistribution<T>(Guid runId, IDataView dataView, string columnName) where T:struct
```

Comment:
For this method I was not really thinking of exposing it to the `DataCatalog`. I was more thinking of making sure that the `DataDistributions` are loaded for the `GetData` method so when you get the data it will also be there :) 

Since `DataDistributions` are not an owned property, just remember that you'll need to get them by Id